src/main/java/org/sagebionetworks/web/client/widget/entity/tabs/DiscussionTab.java

```diff
@@ -19,35 +23,68 @@
 	// TODO: use this token to navigate between threads within the discussion tab
 	String areaToken = null;
 	NewThreadModal newThreadModal;
-
-	private ThreadListWidget discussionListWidget;
+	ThreadListWidget threadListWidget;
+	SynapseAlert synAlert;
+	Forum forum;
+	DiscussionForumClientAsync discussionForumClient;
+	CallbackP<Void> newThreadCallback = new CallbackP<Void>(){
+		@Override
+		public void invoke(Void param) {
+			configureThreadList();
+		}
+	};
 
 	@Inject
 	public DiscussionTab(
 			DiscussionTabView view,
 			Tab tab,
-			ThreadListWidget discussionListWidget,
+			SynapseAlert synAlert,
+			DiscussionForumClientAsync discussionForumClient,
+			ThreadListWidget threadListWidget,
 			NewThreadModal newThreadModal,
 			CookieProvider cookies
 			) {
 		this.view = view;
 		this.tab = tab;
-		this.discussionListWidget = discussionListWidget;
+		this.synAlert = synAlert;
+		this.threadListWidget = threadListWidget;
 		this.newThreadModal = newThreadModal;
+		this.discussionForumClient = discussionForumClient;
 		this.cookies = cookies;
 		tab.configure("Discussion", view.asWidget());
 		view.setPresenter(this);
-		view.setDiscussionList(discussionListWidget.asWidget());
+		view.setThreadList(threadListWidget.asWidget());
 		view.setNewThreadModal(newThreadModal.asWidget());
+		view.setAlert(synAlert.asWidget());
+	}
+
+	public void configureThreadList() {
+		if (forum != null && forum.getId() != null) {
+			threadListWidget.configure(forum.getId());
+		}
 	}
 
 	public void setTabClickedCallback(CallbackP<Tab> onClickCallback) {
 		tab.addTabClickedCallback(onClickCallback);
 	}
 
-	public void configure(String entityId, String entityName) {
+	public void configure(final String entityId,final String entityName) {
 		tab.setEntityNameAndPlace(entityName, new Synapse(entityId, PROJECT_VERSION_NUMBER, EntityArea.DISCUSSION, areaToken));
 		tab.setTabListItemVisible(DisplayUtils.isInTestWebsite(cookies));
+		discussionForumClient.getForumMetadata(entityId, new AsyncCallback<Forum>(){
+
+			@Override
+			public void onFailure(Throwable caught) {
+				synAlert.handleException(caught);
+			}
+
+			@Override
+			public void onSuccess(Forum result) {
+				forum = result;
+				newThreadModal.configure(forum.getId(), newThreadCallback);
```

Comment:
I think you can simplify by defining the callback here.
CallbackP<Void> newThreadCallback = new CallbackP<Void>(){
		@Override
		public void invoke(Void param) {
			threadListWidget.configure(forum.getId());
		}
	};